cmd/fanoutsidecar/main.go

```diff
@@ -125,57 +127,87 @@ func main() {
 	}
 }
 
-func setupConfigMapNoticer(logger *zap.Logger, configUpdated swappable.UpdateConfig) (manager.Manager, error) {
+func setupChannelWatcher(logger *zap.Logger, configUpdated swappable.UpdateConfig) (manager.Manager, error) {
 	mgr, err := manager.New(config.GetConfigOrDie(), manager.Options{})
 	if err != nil {
-		logger.Error("Error starting manager.", zap.Error(err))
+		logger.Error("Error creating new maanger.", zap.Error(err))
 		return nil, err
 	}
-
-	switch configMapNoticer {
-	case cmnfVolume:
-		err = setupConfigMapVolume(logger, mgr, configUpdated)
-	case cmnfWatcher:
-		err = setupConfigMapWatcher(logger, mgr, configUpdated)
-	default:
-		err = fmt.Errorf("need to provide the --config_map_noticer flag (valid values are %s)", configMapNoticerValues())
-	}
-	if err != nil {
+	if err = v1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
+		logger.Error("Error while adding eventing scheme to manager.", zap.Error(err))
```

Comment:
Should this include the scheme?

Reply:
Is the question whether scheme is needed or should it be in this function?
 - Adding this scheme is necessary or else the watch through manager won't work

- I considered whether I should add it here or in the channelwatcher.New function. Looking at rest of the code where controllers are created, I found that schemes are added in cmd/main rather than  inside the controller. So I kept it this way to be consistent with rest of the code.